runtime/drivers/duckdb/duckdb.go

```diff
@@ -481,10 +485,23 @@ func (c *connection) reopenDB(ctx context.Context) error {
 		"SET old_implicit_casting = true", // Implicit Cast to VARCHAR
 	)
 
+	dataDir, err := c.storage.DataDir()
+	if err != nil {
+		return err
+	}
+
 	// We want to set preserve_insertion_order=false in hosted environments only (where source data is never viewed directly). Setting it reduces batch data ingestion time by ~40%.
 	// Hack: Using AllowHostAccess as a proxy indicator for a hosted environment.
 	if !c.config.AllowHostAccess {
-		bootQueries = append(bootQueries, "SET preserve_insertion_order TO false")
+		tempDir, err := c.storage.TempDir()
+		if err != nil {
+			return err
+		}
+		bootQueries = append(bootQueries,
+			"SET preserve_insertion_order TO false",
+			fmt.Sprintf("SET temp_directory = %s", safeSQLString(tempDir)),
+			fmt.Sprintf("SET secret_directory = %s", safeSQLString(filepath.Join(dataDir, ".duckdb", "secrets"))),
```

Comment:
Will this interfere with external table storage or will it just be ignored?

Reply:
Good point. These queries are executed for every db file so setting `secret_directory` is what is expected. 
But there could be issues if all database share same tmp directory. May be for new removing `temp_directory` seems to be a safe option. 
More important was to set unique `secret_directory` for each instance's database.